### ISSUE TITLE
训练模型时model中的acc总是很低41%多，但是val_acc和test_acc能达到87%，是什么原因

### ISSUE THREAD

**Issue by wanyuks** [bojone/bert4keras#222]
提问时请尽可能提供如下信息：

### 基本信息
- 你使用的**操作系统**: win10
- 你使用的**Python**版本: 3.8
- 你使用的**Tensorflow**版本: 2.3
- 你使用的**Keras**版本: 2.43
- 你使用的**bert4keras**版本: 0.87
- 你使用纯**keras**还是**tf.keras**: tf.keras
- 你加载的**预训练模型**:bert

### 核心代码
```python
task_sentence_similarity_lcqmc.py 
```

### 输出信息
```

![image](https://user-images.githubusercontent.com/38677249/94406106-1b58c980-01a4-11eb-8cf9-526accc28844.png)

```

### 自我尝试
不管什么问题，请先尝试自行解决，“万般努力”之下仍然无法解决再来提问。此处请贴上你的努力过程。

win

**Comment by bojone** [comment]
直接跑的`task_sentence_similarity_lcqmc.py`？

**Comment by wanyuks** [comment]
> 直接跑的`task_sentence_similarity_lcqmc.py`？

def read_tsv(input_file, quotechar=None):
    """读取数据"""
    with tf.io.gfile.GFile(input_file, "r") as f:
        next(f)
        reader = csv.reader(f, delimiter="\t", quotechar=quotechar)
        lines = []
        for line in reader:
            text1, text2, label = line[0], line[1], line[2]
            lines.append((text1, text2, int(label)))
        return lines

修改了数据读取模块，然后batch_size=32

**Comment by bojone** [comment]
> > 直接跑的`task_sentence_similarity_lcqmc.py`？
> 
> def read_tsv(input_file, quotechar=None):
> """读取数据"""
> with tf.io.gfile.GFile(input_file, "r") as f:
> next(f)
> reader = csv.reader(f, delimiter="\t", quotechar=quotechar)
> lines = []
> for line in reader:
> text1, text2, label = line[0], line[1], line[2]
> lines.append((text1, text2, int(label)))
> return lines
> 
> 修改了数据读取模块，然后batch_size=32

也就是说这是你自己的数据集？有点神奇，train acc低，valid acc正常，那只能说明train acc写得不对，不清楚哪里不对了（捂脸）

**Comment by wanyuks** [comment]
> > > 直接跑的`task_sentence_similarity_lcqmc.py`？
> > 
> > 
> > def read_tsv(input_file, quotechar=None):
> > """读取数据"""
> > with tf.io.gfile.GFile(input_file, "r") as f:
> > next(f)
> > reader = csv.reader(f, delimiter="\t", quotechar=quotechar)
> > lines = []
> > for line in reader:
> > text1, text2, label = line[0], line[1], line[2]
> > lines.append((text1, text2, int(label)))
> > return lines
> > 修改了数据读取模块，然后batch_size=32
> 
> 也就是说这是你自己的数据集？有点神奇，train acc低，valid acc正常，那只能说明train acc写得不对，不清楚哪里不对了（捂脸）

数据集就是lcqmc的数据，train_acc是tensorflow的模型训练时自己计算的acc，数据集应该和您一样的，然后除了读取数据模块其他代码都是和您一样的，然后我今天用tf1.15试跑了一下，发现结果看起来正常多了，所以怀疑是不是tf2.3的bug？
![image](https://user-images.githubusercontent.com/38677249/94638644-021f5c80-030d-11eb-8353-549fc7899796.png)

**Comment by wanyuks** [comment]
> > > 直接跑的`task_sentence_similarity_lcqmc.py`？
> > 
> > 
> > def read_tsv(input_file, quotechar=None):
> > """读取数据"""
> > with tf.io.gfile.GFile(input_file, "r") as f:
> > next(f)
> > reader = csv.reader(f, delimiter="\t", quotechar=quotechar)
> > lines = []
> > for line in reader:
> > text1, text2, label = line[0], line[1], line[2]
> > lines.append((text1, text2, int(label)))
> > return lines
> > 修改了数据读取模块，然后batch_size=32
> 
> 也就是说这是你自己的数据集？有点神奇，train acc低，valid acc正常，那只能说明train acc写得不对，不清楚哪里不对了（捂脸）

![image](https://user-images.githubusercontent.com/38677249/94638747-41e64400-030d-11eb-8920-60f36bbcb502.png)

**Comment by bojone** [comment]
那就应该是tf 2.3的问题了，你应该用的是keras吧，它可能不大配tf 2.3了。
